pkg/scanners/helm/parser/parser.go

```diff
@@ -0,0 +1,227 @@
+package parser
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/releaseutil"
+
+	"github.com/aquasecurity/defsec/internal/debug"
+	"github.com/aquasecurity/defsec/pkg/detection"
+	"github.com/aquasecurity/defsec/pkg/scanners/options"
+)
+
+var manifestNameRegex = regexp.MustCompile("# Source: [^/]+/(.+)")
+
+type Parser struct {
+	chartName    string
+	helmClient   *action.Install
+	rootPath     string
+	filepaths    []string
+	debug        debug.Logger
+	skipRequired bool
+	workingFS    fs.FS
+}
+
+type ChartFile struct {
+	TemplateFilePath string
+	ManifestContent  string
+}
+
+func (p *Parser) SetDebugWriter(writer io.Writer) {
+	p.debug = debug.New(writer, "parse:helm")
+}
+
+func (p *Parser) SetSkipRequiredCheck(b bool) {
+	p.skipRequired = b
+}
+
+func New(chartName string, options ...options.ParserOption) *Parser {
+
+	client := action.NewInstall(&action.Configuration{})
+	client.DryRun = true  // don't do anything
+	client.Replace = true // skip name check
+	client.ReleaseName = chartName
+	client.ClientOnly = true // don't try to talk to a cluster
+
+	p := &Parser{
+		chartName:  chartName,
+		helmClient: client,
+	}
+
+	for _, option := range options {
+		option(p)
+	}
+	return p
+}
+
+func (p *Parser) ParseFS(ctx context.Context, target fs.FS, path string) error {
+	p.workingFS = target
+
+	if err := fs.WalkDir(p.workingFS, filepath.ToSlash(path), func(path string, entry fs.DirEntry, err error) error {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		if err != nil {
+			return err
+		}
+		if entry.IsDir() {
+			return nil
+		}
+
+		if isArchive(path) {
+			tarFS, err := p.addTarToFS(path)
+			if err != nil {
+				return err
+			}
+			if err := p.ParseFS(ctx, tarFS, "."); err != nil {
+				return err
+			}
+			return nil
+		}
+
+		return p.addPaths(ctx, path)
```

Comment:
I'm not sure it matters, but it looks like the same path will get added multiple times as ParseFS is called again each time a tar is added. Will it cause any problems having duplicate paths in the slice? It doesn't look like it to me apart from being slightly inefficient but I don't think it's going to be noticeable.

Reply:
I don't see the flow where that happens. If the tar is added, it returns after its parsed it, if not, the path is added here.